src/semantic/check_multiple_declaration.cr

```diff
@@ -11,5 +11,13 @@ module Semantic
       @names << definition.name
       super
     end
+    
+    def visit(op : AST::Operation)
+      if @operationsNames.includes? op.name
+        raise SemanticException.new("Function '#{op.name}' is declared multiples times")
+      end
+      @operationsNames << op.name
```

Comment:
1. Better name it `@operationNames`
2. You didn't declate it. This won't compile.